lib/Optimizer/Builder/Intrinsics.cpp

```diff
@@ -47,8 +47,33 @@ inline bool operator<(const IntrinsicCode &icode, const IntrinsicCode &jcode) {
 /// well as prototypes for LLVM intrinsics and C library calls that are used by
 /// the compiler. The table should be kept in sorted order.
 static constexpr IntrinsicCode intrinsicTable[] = {
+    // Initialize a (preallocated) buffer (the first parameter) with i64 values
```

Comment:
Wow, I hadn't noticed this file before, and it looks interesting. Anyway, the comment above says that it should be sorted, but it doesn't say how. I assume its alphabetical based the first item, but is there anything enforcing that?  It looks like it is using `cudaq::setCudaqRangeVector`, which could be anything. I guess I'll just ask the dumb question: what happens if it isn't sorted correctly? If the answer is: it throws an error message, then that's good, but if it silently misbehaves, maybe we need additional comments where the `cudaq::setCudaqRangeVector` value is defined, so that people know not to change it without considering this order?

Reply:
Good questions.

- Yes, alphabetical order.

- Some of the map keys are given explicit names in the header file. That is meant to follow a one-definition sort of rule, so if the key ever needs to be changed no one has to grep for string content. (It should probably be the case for all the keys, but alas.

- Yes, there is a live check that determines if the entries in the table are in alphabetical order. If not, the compiler gives an error message and terminates, so it's pretty obvious. (And, yeah, it's easy to change the name, say, and forget to move the row in the table. Hence the check.)